runtime/compiler/pandoc.vim

```diff
@@ -0,0 +1,23 @@
+" Vim compiler file
+" Compiler:     Pandoc
+" Maintainer:   Konfekt
```

Comment:
@Konfekt are you fine with being the maintainer here?

Reply:
Yes, that's fine

Reply:
@Freed-Wu I updated https://github.com/Konfekt/vim-compilers/blob/master/compiler/pandoc.vim to infer the from format from the file type (in case one edits `rst`, `latex`, ... files, or other less common formats, also accepted by pandoc).

Reply:
... updated again, for what it's worth, now detecting more input formats